internal/p2p/peermanager.go

```diff
@@ -1389,3 +1414,31 @@ func keyPeerInfoRange() ([]byte, []byte) {
 	}
 	return start, end
 }
+
+// EvictPeer evicts a peer by a node id
+func (m *PeerManager) EvictPeer(nodeID types.NodeID) {
+	m.mtx.Lock()
+	m.evict[nodeID] = true
+	m.mtx.Unlock()
+	m.evictWaker.Wake()
+}
+
+// UpdatePeerInfo modifies a peer-info using a function modificator, this operation is a transactional
+func (m *PeerManager) UpdatePeerInfo(nodeID types.NodeID, modifier func(peerInfo peerInfo) peerInfo) error {
+	// peerManager.store assumes that peerManager is managing it
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+	peer, ok := m.store.Get(nodeID)
+	if !ok {
+		return errPeerNotFound(fmt.Errorf("peer with id %s not found", nodeID))
+	}
+	modifier(peer)
```

Comment:
not sure if I understand correctly, but the modifier is defined as:
```
func(peerInfo peerInfo) peerInfo
```
so, it's not changing the arg, but returning updated arg instead.

Here we lose returned value, so the `modifier` will most likely do nothing.